apps/src/storage/dataBrowser/DataTableView.jsx

```diff
@@ -0,0 +1,232 @@
+/**
+ * @overview Component for detailed view of a data table.
+ */
+import TableControls from './TableControls';
+import {DataView} from '../constants';
+import DataTable from './DataTable';
+import FirebaseStorage from '../firebaseStorage';
+import FontAwesome from '../../templates/FontAwesome';
+import PropTypes from 'prop-types';
+import Radium from 'radium';
+import React from 'react';
+import {changeView, showWarning} from '../redux/data';
+import * as dataStyles from './dataStyles';
+import color from '../../util/color';
+import {connect} from 'react-redux';
+import {getColumnNamesFromRecords} from '../firebaseMetadata';
+import experiments from '../../util/experiments';
+
+const MIN_TABLE_WIDTH = 600;
+
+const styles = {
+  addColumnHeader: [
+    dataStyles.headerCell,
+    {
+      width: 19
+    }
+  ],
+  container: {
+    flexDirection: 'column',
+    height: '100%',
+    minWidth: MIN_TABLE_WIDTH,
+    maxWidth: '100%',
+    paddingLeft: experiments.isEnabled(experiments.APPLAB_DATASETS)
+      ? '8px'
+      : '0px'
+  },
+  table: {
+    minWidth: MIN_TABLE_WIDTH
+  },
+  tableWrapper: {
+    flexGrow: 1,
+    overflow: 'scroll'
+  },
+  pagination: {
+    float: 'right',
+    display: 'inline',
+    marginTop: 10
+  },
+  plusIcon: {
+    alignItems: 'center',
+    borderRadius: 2,
+    backgroundColor: 'white',
+    color: color.teal,
+    cursor: 'pointer',
+    display: 'inline-flex',
+    height: 18,
+    justifyContent: 'center',
+    width: 18
+  }
+};
+
+const INITIAL_STATE = {
+  showDebugView: false
+};
+
+class DataTableView extends React.Component {
+  static propTypes = {
+    // from redux state
+    tableColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+    tableName: PropTypes.string.isRequired,
+    // "if all of the keys are integers, and more than half of the keys between 0 and
+    // the maximum key in the object have non-empty values, then Firebase will render
+    // it as an array."
+    // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
```

Comment:
What is this comment referring to? tableName?

Reply:
`tableRecords` - it explains why `tableRecords` is either an object or an array. Same comment as currently at https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/storage/dataBrowser/DataTable.jsx#L81

Reply:
Ahh makes sense.